tests/test_split_corrnoise_basis_weights.py

```diff
@@ -0,0 +1,88 @@
+"""Test if the split basis and weights functions for EcorrNoise and PLRedNoise
+give the same result as the old code."""
+
+import numpy as np
+import astropy.units as u
+import pytest
+import warnings
+from pint.config import examplefile
+from pint.models import get_model_and_toas
+from pint.models.noise_model import (
+    create_ecorr_quantization_matrix,
+    create_fourier_design_matrix,
+    get_rednoise_freqs,
+    powerlaw,
+)
+
+
+@pytest.fixture()
+def model_and_toas():
+    parfile = examplefile("B1855+09_NANOGrav_9yv1.gls.par")
+    timfile = examplefile("B1855+09_NANOGrav_9yv1.tim")
```

Comment:
This file has a lot of TOAs and takes a long time to load, every time the test suite is run. Is there a smaller example that would work?

Reply:
I'll create a smaller version of this.